src/StreamJsonRpc/MessagePackFormatter.cs

```diff
@@ -2140,6 +2192,7 @@ public unsafe void Serialize(ref MessagePackWriter writer, TraceParent value, Me
         private class JsonRpcRequest : Protocol.JsonRpcRequest, IJsonRpcMessageBufferManager, IJsonRpcMessagePackRetention
         {
             private readonly MessagePackFormatter formatter;
+            private Dictionary<string, ReadOnlySequence<byte>>? topLevelProperties;
 
```

Comment:
If you type this as `Sequence<byte>`, then after you transmit the message you could clear the dictionary and `Dispose` of all the values in order to recycle the buffers used for these.

Reply:
I don't see Sequence being as disposable, do you mean just clear the dictionary contents after serializing the message? If so I think the new changes might achieve that.

Reply:
> `Sequence<T>` [is definitely disposable](http://fuget.org/packages/Nerdbank.Streams/2.8.46/lib/netstandard2.0/Nerdbank.Streams.dll/Nerdbank.Streams/Sequence%601).

We might have two dictionaries: one for outbound that stores `Sequence<T>` so it can be disposed of, and one for inbound that stores `ReadOnlySequence<T>` whose buffers are already recycled at the conclusion of deserialization.

Reply:
The sequences are released now with the latest changes from both of us :)